pkg/controllers/nodeBackend/nodeBackend.go

```diff
@@ -0,0 +1,496 @@
+package nodeBackend
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/oracle/oci-go-sdk/v65/common"
+	"github.com/oracle/oci-native-ingress-controller/pkg/client"
+	"github.com/oracle/oci-native-ingress-controller/pkg/controllers/ingressclass"
+
+	"k8s.io/klog/v2"
+
+	ociloadbalancer "github.com/oracle/oci-go-sdk/v65/loadbalancer"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	networkinginformers "k8s.io/client-go/informers/networking/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	networkinglisters "k8s.io/client-go/listers/networking/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/oracle/oci-native-ingress-controller/pkg/util"
+)
+
+var errIngressClassNotReady = errors.New("ingress class not ready")
+
+const (
+
+	// ServiceAnnotationLoadBalancerNodeFilter is a service annotation to select specific nodes as your backend in the LB
+	// based on label selector.
+	ServiceAnnotationLoadBalancerNodeFilter = "oci.oraclecloud.com/node-label-selector"
+
+	// ToBeDeletedTaint is a taint used by the CLuster Autoscaler before marking a node for deletion. Defined in
+	// https://github.com/kubernetes/autoscaler/blob/e80ab518340f88f364fe3ef063f8303755125971/cluster-autoscaler/utils/deletetaint/delete.go#L36
+	ToBeDeletedTaint = "ToBeDeletedByClusterAutoscaler"
+)
+
+type Controller struct {
+	controllerClass string
+
+	ingressClassLister networkinglisters.IngressClassLister
+	ingressLister      networkinglisters.IngressLister
+	serviceLister      corelisters.ServiceLister
+	podLister          corelisters.PodLister
+	nodeLister         corelisters.NodeLister
+	endpointLister     corelisters.EndpointsLister
+
+	queue workqueue.RateLimitingInterface
+
+	client *client.ClientProvider
+}
+
+func NewController(controllerClass string, ingressClassInformer networkinginformers.IngressClassInformer, ingressInformer networkinginformers.IngressInformer, serviceLister corelisters.ServiceLister, endpointLister corelisters.EndpointsLister, podLister corelisters.PodLister, nodeLister corelisters.NodeLister,
+	client *client.ClientProvider) *Controller {
+
+	c := &Controller{
+		controllerClass:    controllerClass,
+		ingressClassLister: ingressClassInformer.Lister(),
+		ingressLister:      ingressInformer.Lister(),
+		serviceLister:      serviceLister,
+		endpointLister:     endpointLister,
+		podLister:          podLister,
+		nodeLister:         nodeLister,
+		client:             client,
+		queue:              workqueue.NewRateLimitingQueue(workqueue.NewItemExponentialFailureRateLimiter(10*time.Second, 5*time.Minute)),
+	}
+
+	return c
+}
+
+func (c *Controller) processNextItem() bool {
```

Comment:
Refactor this function (any other similar functions). This is repeated in all controllers.

Reply:
Done for both the backend controllers.